CoreScriptsRoot/Modules/VR/NotificationHub.lua

```diff
@@ -0,0 +1,732 @@
+
+				BackgroundTransparency = 1,
+
+				Image = BUTTON_NORMAL_IMG,
+				ScaleType = Enum.ScaleType.Slice,
+				SliceCenter = Rect.new(8,6,46,44),
+
+				SelectionImageObject = emptySelectionImage
+			}, Util:Create "TextLabel" {
+				Position = UDim2.new(0, 0, 0, 0),
+				Size = UDim2.new(1, 0, 1, -6),
+
+				BackgroundTransparency = 1,
+
+				Text = text,
+				TextColor3 = TEXT_COLOR,
+				Font = Enum.Font.SourceSansBold,
+				FontSize = Enum.FontSize.Size24
+			}
+			text.Parent = button
+
+			button.SelectionGained:connect(function()
+				button.Image = BUTTON_SELECTED_IMG
+			end)
+			button.SelectionLost:connect(function()
+				button.Image = BUTTON_NORMAL_IMG
+			end)
+			return button, text
+		end
+
+		if notificationInfo.Button1Text and notificationInfo.Button2Text then
+			self.detailsButton1, self.detailsButton1Text = createButton(0.07, 0.415, notificationInfo.Button1Text)
+			self.detailsButton2, self.detailsButton2Text = createButton(0.511, 0.415, notificationInfo.Button2Text)
+
+			self.detailsButton1.MouseButton1Click:connect(function() 
+				doCallback(notificationInfo.Callback, notificationInfo.Button1Text)
+				self:Dismiss() 
+			end)
+			self.detailsButton2.MouseButton1Click:connect(function() 
+				doCallback(notificationInfo.Callback, notificationInfo.Button2Text)
+				self:Dismiss() 
+			end)
+		elseif not notificationInfo.button2Text then
+			local text = notificationInfo.Button1Text or "Dismiss"
+			self.detailsButton1, self.detailsButton1Text = createButton(0.07, 0.86, text)
+			self.detailsButton1.MouseButton1Click:connect(function() 
+				doCallback(notificationInfo.Callback, notificationInfo.Button1Text)
+				self:Dismiss() 
+			end)
+		end
+
+		table.insert(group.notifications, self)
+		group.notificationsDirty = true
+		layoutNotificationsGroups()
+
+		return self
+	end
+
+	function Notification:OnClicked()
+		self.group:BringNotificationToFront(self)
+	end
+
+	function Notification:Dismiss()
+		self.group:RemoveNotification(self)
+	end
+end
+
+--NotificationHubModule API and state management
+do
+	local pendingNotifications = {}
+	local isVisible = false
+	local unreadCount = 0
+
+	local SendNotificationInfoEvent = RobloxGui:WaitForChild("SendNotificationInfo")
+	SendNotificationInfoEvent.Event:connect(function(notificationInfo)
+		local group = notificationsGroups[notificationInfo.GroupName or "Developer"]
+		if not group then
+			group = notificationsGroups.Developer
+		end
+
+		Notification.new(group, notificationInfo)
+
+		if not isVisible then
+			unreadCount = unreadCount + 1
+			NotificationHubModule.UnreadCountChanged(unreadCount)
+		end
+	end)
+
+	NotificationHubModule.VisibilityStateChanged = Util:Create "BindableEvent" {
+		Name = "VisibilityStateChanged"
+	}
+
+	function NotificationHubModule:GetNumberOfPendingNotifications()
+		return #pendingNotifications
+	end
+
+	function NotificationHubModule:IsVisible()
+		return isVisible
+	end
+
+	local cancelAnimation = false
```

Comment:
looks like cancelAnimation is only assigned to(line 676 and 705) but never used for any evaluation.